### DISCUSSION TITLE
Capo Element UX issues persist from MuseScore 4.1 to 4.2

### DISCUSSION THREAD

**Discussion by scorster** [musescore/MuseScore#25578] [General]
### Issue type

UX/Interaction bug (incorrect behaviour)

### Bug description

Please see forum post: https://musescore.org/en/node/359183

### Steps to reproduce

    • Open MuseScore 4
    • Add a Capo element to a measure
    • Try to select, drag or delete

RESULT: Doesn't work because clicking the element opens it.

### MuseScore Version

MuseScore 4.2

### Regression

No.

### Operating system

MacOS 13.5.2

**Comment by MarcSabatella** [comment]
You can certainly do all of those things despite the presence of the dialog.  Click and it is selected.  Use the cursor keys or Properties panel or drag and it moves.  Press Delete and it deletes.

Definitely room for improvement in the interaction - for instance, it would be nice if Esc left the capo selected after closing the dialog, the same way clicking any other GUI element does.  But it's simply not true to say you can't select, move, or delete it.

**Comment by scorster** [comment]
Hi Mark, 

Thanks. It was helpful to know that you can select, drag and delete a Capo element on you system! Perhaps the behaviors I'm encountering are MacOS specific.

In preparing to make a video demonstrating the capo element not selectable on my Mac I discovered that the Capo element is indeed selectable (which then makes it draggable and delectable) *BUT* the selection method is peculiar.

Unlike your description "Click and it is selected" ... here clicking the capo element always opens the dialog, as I stated in the original post. (Assuming we agree that I click" is a quick mouseDown/MouseUp.) Whereas in MuseScore UX we select with a click, and the object is highlighted/selected; and generally we can drag an unselected object.

### The source of the confusion

After dismissing the capo element popup (either by pressing esc or clicking outside of it) the capo element becomes deselected. But a subsequent click opens it. And that's the cycle I was stuck in. 

I'd expect I could simply drag the unselected capo element to reposition it—just like we do with other object, like staff text—but there's not chance to drag because mouseDown opens the popup.

Never occurred to me to click on the highlighted/selected capo element *while the popup was open.* Doing so closes the popup and leaves the element selected, thus draggable and deleteable. 

### Here are the behaviors I see:

• When a capo element is unhighlighted/ unselected a single click or mouseDown opens the Capo popup and MuseScore and Capo element shows as selected. 

• Attempts to drag an unselected Capo element fail because they open the popup. Attempts to select the element without opening the popup are not possible because the dialog opens on mouseDown.

• Esc. closes the popup and deselects the capo element (As you said, *"... it would be nice if Esc left the capo selected after closing the popup, the same way clicking any other GUI element does."* I completely agree about that, probably would have never lost my way!

• Clicking outside the popup also closes the popup and deselects the capo element. (Maybe it should stay selected, as with the proposed Esc behavior.)

So the only way to drag the element is to drag it while its selected. That's true:

• only when the capo element is open
• Or ODDLY, but double clicking the element while the capo popup is open.

That's some weird and whacky stuff.

### The important questions are:

• Is this by design?
• Is it in keeping with the overarching MuseScore UX
• Does ANY OTHER dialog in MuseScore behave this way? 

### Questions

Is the situation fully normal on your end, Mark? Perhaps we need feedback from another Mac user. Let me know if you want me to make a video.

**Comment by MarcSabatella** [comment]
For me on Linux, clicking both selects it and opens the popup.  So it is not at all inaccurate to say "click and it is selected".  It's true something *else* happens as well, but as far as I can see the presence of the popup does not interfere with normal behavior in any way whatsoever.  All actions that can be performed after selecting any other object also work on the capo.  At least, no issues I can find.  Basically everything works exactly as it would if the popup weren't there, with only one exception - dragging requires a second click.  As far as I know this is all by design.

As I mentioned, clicking outside the popup leaves the element selected, *if* you click a GUI element (even an empty spot on the toolbar).  But clicking within the score does what it always does - selects if clicking an element, clears selection if clicking an empty area.

As for what other elements have similar popups, the harp pedal diagram popup is similar, and so is the guitar tuning popup.  There has been talk of adding popups to other elements for easier access to their properties as well.

**Comment by rgreen5** [comment]
I have to agree that automatic opening of the Tuning text when you click on it is a bit discombobulating. IMV it would be better if stayed closed until the user applied an edit command to it: e.g. an option from a context menu?

**Comment by ShibuyaCyana** [comment]
I don't like this behavior either, and wrote a forum post about it...#20432
>Does ANY OTHER dialog in MuseScore behave this way?

Yes, the harp pedal diagram and the new tuning window - and text property will likely join in a few updates.

**Comment by MarcSabatella** [comment]
> it would be better if stayed closed until the user applied an edit command to it: e.g. an option from a context menu?

I believe the whole point is to avoid going back to the days where important options are hidden behind hard-to-discover controls like context menus.  Especially on Mac, many users have no idea how to access those.  Double-click (with Enter or F2 as shortcut) could be fine though, since it's pretty universal.

**Comment by scorster** [comment]
> ... the whole point is to avoid [having] important options are hidden behind hard-to-discover controls like context menus. 

Right-click contextual menus. Those are one of the *first* places I look for options and properties. But in MuseScore 4 even **Add>Measures** and **Add>Frames** and the entire **Add menu** are removed from Staff>contextual menu. 

> Especially on Mac, many users have no idea how to access those.

Mmm ... because you believe that all Mac are unaware that Control-click is the equivalent to Right-click? But even if a majority of Mac users fall in that category MusesScore documentation could enlighten them.

> Double-click (with Enter or F2 as shortcut) could be fine though, since it's pretty universal.

Agreed. Double-click is universal on desktop apps for accessing properties or opening dialogs. And that's been one my main points. Double-click is familiar and works fine, whereas many people have reported confusion with the single-click popups.

scorster

**Comment by scorster** [comment]
Two other issues with the Capo element popup:

1) there's no option for repositioning the popup window (as we usually can with an old fashioned dialog)
2) a capo element  become inaccessible if dragged into the boundary of a frame. Once fully "inside" a frame it becomes non-clickable, so it can't be moved or deleted. This is likely remedied by setting its z axis forward of the frames.

**Comment by MarcSabatella** [comment]
I didn't say all Mac users are unaware of Ctrl+click.  I said many Mac users don't know about this, based on the very many questions we get over the years (I've responded to many thousands of support questions over the years, as I'm sure you are aware).  I assume this is especially an issue on Mac because with only a one-button mouse, most designers for software for macOS specifically don't tend to bury options in context menus very often, so many Mac users simply never encounter this.

As for context menus in general, it's nice sometimes to have that context menus as an additional way of getting something done that might otherwise take more steps.  It should never be the *primary* way of getting anything done.  note sure what you mean about Add - certainly don't need a context button to add things when there is a button right there on the toolbar.  In any case, unrelated to this issue  Feel free to start a thread on the forum to discuss that further.

Anyhow, requiring people to have to read documentation to learn how to do for things that *can* be made more discoverable is *terrible* UX practice.  I think you'll find that is not up for debate.  But fine details of exactly how the popup is positioned can always be tweaked - that's fair game for discussion of course.

And then I recommend rewriting your original issue description in light of this - focusing on the things that are actual objective issues.  As it is, the original description is simply a misunderstanding, but there *is* good information o be found within the discussion.  That should summarized and moved to the issue description, and then the additional discussion can potentially be hidden.
